src/main/java/net/juniper/netconf/NetconfSession.java

```diff
@@ -61,24 +63,29 @@
     private String lastRpcReply;
     private final DocumentBuilder builder;
     private int messageId = 0;
+    private int commandTimeout;
+
+    public static final int BUFFER_SIZE = 8 * 1024;
 
     private static final String CANDIDATE_CONFIG = "candidate";
     private static final String EMPTY_CONFIGURATION_TAG = "<configuration></configuration>";
     private static final String RUNNING_CONFIG = "running";
     private static final String NETCONF_SYNTAX_ERROR_MSG_FROM_DEVICE = "netconf error: syntax error";
 
-    NetconfSession(Channel netconfChannel, int timeout, String hello,
+    NetconfSession(Channel netconfChannel, int connectionTimeout, int commandTimeout,
+                   String hello,
                    DocumentBuilder builder) throws IOException {
```

Comment:
do you have an overloaded method to make one of these with the old signature? The github diff is not making it obvious.

It would be nice to have both, plus it would be nice to have a builder to make one of these, but that is for a future feature pr. :)